### ISSUE TITLE
Norm layer weight decay not set as expected

### ISSUE THREAD

**Issue by gathierry** [facebookresearch/detectron2#737]
## Instructions To Reproduce the Issue:

1. what changes you made (`git diff`) or what code you wrote

add
```
print(key)
```
just after https://github.com/facebookresearch/detectron2/blob/0bf1ae2cd1b87ae9e61732776ccd816bc5a16ae7/detectron2/solver/build.py#L15
to print all keys in `model.named_paramters()`

2. what exact command you run:

```bash
python tools/train_net --config-file /Users/shiyuli/Documents/detectron2/configs/COCO-Detection/faster_rcnn_R_50_FPN_1x.yaml
```

3. what you observed (including the full logs):
Here's everything printed
```
backbone.fpn_lateral2.weight
backbone.fpn_lateral2.bias
backbone.fpn_output2.weight
backbone.fpn_output2.bias
backbone.fpn_lateral3.weight
backbone.fpn_lateral3.bias
backbone.fpn_output3.weight
backbone.fpn_output3.bias
backbone.fpn_lateral4.weight
backbone.fpn_lateral4.bias
backbone.fpn_output4.weight
backbone.fpn_output4.bias
backbone.fpn_lateral5.weight
backbone.fpn_lateral5.bias
backbone.fpn_output5.weight
backbone.fpn_output5.bias
backbone.bottom_up.stem.conv1.weight
backbone.bottom_up.res2.0.shortcut.weight
backbone.bottom_up.res2.0.conv1.weight
backbone.bottom_up.res2.0.conv2.weight
backbone.bottom_up.res2.0.conv3.weight
backbone.bottom_up.res2.1.conv1.weight
backbone.bottom_up.res2.1.conv2.weight
backbone.bottom_up.res2.1.conv3.weight
backbone.bottom_up.res2.2.conv1.weight
backbone.bottom_up.res2.2.conv2.weight
backbone.bottom_up.res2.2.conv3.weight
backbone.bottom_up.res3.0.shortcut.weight
backbone.bottom_up.res3.0.conv1.weight
backbone.bottom_up.res3.0.conv2.weight
backbone.bottom_up.res3.0.conv3.weight
backbone.bottom_up.res3.1.conv1.weight
backbone.bottom_up.res3.1.conv2.weight
backbone.bottom_up.res3.1.conv3.weight
backbone.bottom_up.res3.2.conv1.weight
backbone.bottom_up.res3.2.conv2.weight
backbone.bottom_up.res3.2.conv3.weight
backbone.bottom_up.res3.3.conv1.weight
backbone.bottom_up.res3.3.conv2.weight
backbone.bottom_up.res3.3.conv3.weight
backbone.bottom_up.res4.0.shortcut.weight
backbone.bottom_up.res4.0.conv1.weight
backbone.bottom_up.res4.0.conv2.weight
backbone.bottom_up.res4.0.conv3.weight
backbone.bottom_up.res4.1.conv1.weight
backbone.bottom_up.res4.1.conv2.weight
backbone.bottom_up.res4.1.conv3.weight
backbone.bottom_up.res4.2.conv1.weight
backbone.bottom_up.res4.2.conv2.weight
backbone.bottom_up.res4.2.conv3.weight
backbone.bottom_up.res4.3.conv1.weight
backbone.bottom_up.res4.3.conv2.weight
backbone.bottom_up.res4.3.conv3.weight
backbone.bottom_up.res4.4.conv1.weight
backbone.bottom_up.res4.4.conv2.weight
backbone.bottom_up.res4.4.conv3.weight
backbone.bottom_up.res4.5.conv1.weight
backbone.bottom_up.res4.5.conv2.weight
backbone.bottom_up.res4.5.conv3.weight
backbone.bottom_up.res5.0.shortcut.weight
backbone.bottom_up.res5.0.conv1.weight
backbone.bottom_up.res5.0.conv2.weight
backbone.bottom_up.res5.0.conv3.weight
backbone.bottom_up.res5.1.conv1.weight
backbone.bottom_up.res5.1.conv2.weight
backbone.bottom_up.res5.1.conv3.weight
backbone.bottom_up.res5.2.conv1.weight
backbone.bottom_up.res5.2.conv2.weight
backbone.bottom_up.res5.2.conv3.weight
proposal_generator.rpn_head.conv.weight
proposal_generator.rpn_head.conv.bias
proposal_generator.rpn_head.objectness_logits.weight
proposal_generator.rpn_head.objectness_logits.bias
proposal_generator.rpn_head.anchor_deltas.weight
proposal_generator.rpn_head.anchor_deltas.bias
roi_heads.box_head.fc1.weight
roi_heads.box_head.fc1.bias
roi_heads.box_head.fc2.weight
roi_heads.box_head.fc2.bias
roi_heads.box_predictor.cls_score.weight
roi_heads.box_predictor.cls_score.bias
roi_heads.box_predictor.bbox_pred.weight
roi_heads.box_predictor.bbox_pred.bias
```

4. please also simplify the steps as much as possible so they do not require additional resources to
	 no additional resource required

## Expected behavior:

Normally there should be some keys ends with `norm.weight` or `norm.bias` so that we can set them to 0. But it turns out there's no such keys and the corresponding code is never called.

## Environment:

```
---------------------  -------------------------------------------------------------------------------------------------------------------
sys.platform           darwin
Python                 3.6.2 |Anaconda, Inc.| (default, Oct  5 2017, 03:00:07) [GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]
Numpy                  1.17.4
detectron2._C          failed to import
DETECTRON2_ENV_MODULE  <not set>
PyTorch                1.4.0
PyTorch Debug Build    False
torchvision            unknown
CUDA available         False
Pillow                 6.2.2
cv2                    3.4.2
---------------------  -------------------------------------------------------------------------------------------------------------------
PyTorch built with:
  - GCC 4.2
  - clang 9.0.0
  - Intel(R) Math Kernel Library Version 2019.0.5 Product Build 20190808 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
  - NNPACK is enabled
  - Build settings: BLAS=MKL, BUILD_NAMEDTENSOR=OFF, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -Wno-deprecated-declarations -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -Wno-invalid-partial-specialization -Wno-typedef-redefinition -Wno-unknown-warning-option -Wno-unused-private-field -Wno-inconsistent-missing-override -Wno-aligned-allocation-unavailable -Wno-c++14-extensions -Wno-constexpr-not-const -Wno-missing-braces -Qunused-arguments -fcolor-diagnostics -faligned-new -fno-math-errno -fno-trapping-math -Wno-unused-private-field -Wno-missing-braces -Wno-c++14-extensions -Wno-constexpr-not-const, DISABLE_NUMA=1, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=OFF, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=ON, USE_OPENMP=OFF, USE_STATIC_DISPATCH=OFF,
```

**Comment by gathierry** [comment]
Is this because all BatchNorm layers are frozen and there's no more other normalization layers in RPN and RCNN heads?
Does that mean one must use the `Conv2d` provided in `layers/wrappers` which has an attribute named `norm`, to adapt to the build optimizer code?

**Comment by ppwwyyxx** [comment]
> Is this because all BatchNorm layers are frozen and there's no more other normalization layers in RPN and RCNN heads?

Yes

> Does that mean one must use the Conv2d provided in layers/wrappers which has an attribute named norm, to adapt to the build optimizer code?

No. However, do note that layers with name "norm" will have a different weight decay.

**Comment by gathierry** [comment]
Okay, thank you for the explanation.
